File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function (matrix) {
    // for rows
    let col = []
    for (let i = 0; i < matrix.length; i++) {
        if (matrix[i].includes(0)) {
            for (let j = 0; j < matrix[i].length; j++) {
                if (matrix[i][j] == 0) {
                    col.push(j)
                }
                matrix[i][j] = 0
            }
        }
    }

    for (let i = 0; i < matrix.length; i++) {
        for (let j of col) {
            matrix[i][j] = 0
        }
    }
    return matrix

};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        row = len(matrix)
        col = len(matrix[0])
        rows = []
        cols = []
        # for columns
        for i in range(col):
            for j in range(row):
                if matrix[j][i] == 0:
                    cols.append(i)
        # for rows
        for i in range(0,row):
            if 0 in matrix[i]:
                matrix[i] = [0] * len(matrix[i])
                
        for i in cols:
            for j in range(row):
                matrix[j][i] = 0
        return matrix


        